File: 0344-reverse-string/0344-reverse-string.java
class Solution {
    public void reverseString(char[] s) {
        reverse(0, s);
        
    }
    
    public void reverse(int index, char[] s){
        if(s==null || index>=s.length){ return; }
        else{
            int l=0, r=s.length-1;
            while(l<r){
                char temp;
                temp=s[l];
                s[l]=s[r];
                s[r]=temp;
                l++;
                r--;
            }
        }
    }
}

File: 0344-reverse-string/0344-reverse-string.cpp
class Solution {
public:
    void reverseString(vector<char>& s) {
            // Two pointers approach
        int left=0;
        int right=s.size()-1;
        
        while(left<right){
            char temp=s[left];
            s[left]=s[right];
            s[right]=temp;
            left++;
            right--;
        }
    }
};